rust/tests/optimize_test.rs

```diff
@@ -0,0 +1,344 @@
+mod optimize {
+    extern crate deltalake;
+
+    use arrow::datatypes::Schema as ArrowSchema;
+    use arrow::{
+        array::{Int32Array, StringArray},
+        datatypes::{DataType, Field},
+        record_batch::RecordBatch,
+    };
+    use deltalake::optimize::{MetricDetails, Metrics};
+    use deltalake::writer::DeltaWriterError;
+    use deltalake::{
+        action, get_backend_for_uri_with_options,
+        optimize::Optimize,
+        writer::{DeltaWriter, RecordBatchWriter},
+        DeltaTableConfig, DeltaTableMetaData, PartitionFilter,
+    };
+    use deltalake::{DeltaTable, Schema, SchemaDataType, SchemaField};
+    use rand::prelude::*;
+    use serde_json::{Map, Value};
+    use std::{collections::HashMap, error::Error, sync::Arc};
+    use tempdir::TempDir;
+    struct Context {
+        pub tmp_dir: TempDir,
+        pub table: DeltaTable,
+    }
+
+    async fn setup_test(partitioned: bool) -> Result<Context, Box<dyn Error>> {
+        let schema = Schema::new(vec![
+            SchemaField::new(
+                "x".to_owned(),
+                SchemaDataType::primitive("integer".to_owned()),
+                false,
+                HashMap::new(),
+            ),
+            SchemaField::new(
+                "y".to_owned(),
+                SchemaDataType::primitive("integer".to_owned()),
+                false,
+                HashMap::new(),
+            ),
+            SchemaField::new(
+                "date".to_owned(),
+                SchemaDataType::primitive("string".to_owned()),
+                false,
+                HashMap::new(),
+            ),
+        ]);
+
+        let p = if partitioned {
+            vec!["date".to_owned()]
+        } else {
+            vec![]
+        };
+
+        let table_meta = DeltaTableMetaData::new(
+            Some("opt_table".to_owned()),
+            Some("Table for optimize tests".to_owned()),
+            None,
+            schema.clone(),
+            p,
+            HashMap::new(),
+        );
+
+        let tmp_dir = tempdir::TempDir::new("opt_table").unwrap();
+        let p = tmp_dir.path().to_str().to_owned().unwrap();
+
+        let backend = get_backend_for_uri_with_options(&p, HashMap::new())?;
+        let mut dt = DeltaTable::new(&p, backend, DeltaTableConfig::default())?;
+        let mut commit_info = Map::<String, Value>::new();
+
+        let protocol = action::Protocol {
+            min_reader_version: 1,
+            min_writer_version: 2,
+        };
+
+        commit_info.insert(
```

Comment:
Seems like we are not using this commit info?

Reply:
Fixed!